File: problems/tree/0226-invert-binary-tree.py
"""#226 Invert Binary Tree
Swap both children at every node while recursively inverting the subtrees.
Time: O(n) | Space: O(h) call stack.
"""


class Solution:
    def invertTree(self, root):
        if root is None:
            return None
        root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)
        return root

File: problems/tree/0226-invert-binary-tree.js
/**
 * #226 Invert Binary Tree
 * Pattern: recursively swap the two children at every node.
 * Time: O(n) | Space: O(h) call stack
 *
 * @param {TreeNode|null} root
 * @return {TreeNode|null}
 */
function invertTree(root) {
  if (root === null) return null

  const originalLeft = root.left
  root.left = invertTree(root.right)
  root.right = invertTree(originalLeft)
  return root
}

module.exports = invertTree
